client.js

```diff
@@ -317,14 +316,14 @@ DHT.prototype._put = function (opts, cb) {
   var isMutable = opts.k
   var hash = isMutable
     ? sha1(opts.salt ? Buffer.concat([ opts.salt, opts.k ]) : opts.k)
-    : sha1(opts.v)
+    : sha1(bencode.encode(opts.v))
```

Comment:
this was one of the things we didnt realize until the end that made it finally work (by re-reading the spec a couple of times)